### DISCUSSION TITLE
Base data on iterations/Sprints

### DISCUSSION THREAD

**Discussion by DrFish22** [nbrown02/FlowViz#44] [Ideas]
Would it be possible to allow data to be analysed per iteration rather than per week?

**Comment by nbrown02** [comment]
So it's not something I would change the template to do, however it can be done :)

There is an iterations table in the back end, you could join this to your dataset, then filter by iteration. I can include some screenshots if you'd like guidance on how to do this...

**Comment by nbrown02** [comment]
@DrFish22 - in the latest version I have made this possible for you, although it won't be configured by default. You will have to drag and drop 'IterationName' either into your slicer at the top (or add a new slicer) and/or drag/drop to replace the 'Week Start Date' value in the relevant chart. Here is an example of it in practice:

![](https://raw.githubusercontent.com/nbrown02/FlowViz/main/Screenshots/Iterations.gif)

Is this the kind of thing you are after?

**Reply by DrFish22** [reply]
Looks great, will try and give it a proper go soon. The project I really want to test it on in ADO doesnt have the right setup currently annoyingly and it looks like everyone who can fix it has gone away for Christmas. Should be able to give you some feedback in the New Year.

**Reply by nbrown02** [reply]
Great stuff - yes please do, it's also made my wiki documentation a bit clearer now so thank you for this feedback/suggestion 👍

**Reply by ImEdmond** [reply]
@nbrown02 We can rest assured that AI will not take over the world because AI was not able to help me with that. Please, how can I see the Fields in order to get the Interactions? If Fields are now called Data what would be the Iteration field to pic
![image](https://github.com/user-attachments/assets/5ac65cb1-e66e-4da0-bc74-49f6875735b6)

**Reply by nbrown02** [reply]
Hi - this looks like you are using the Jira version not the ADO version? I have no idea how to get it for Jira I'm afraid.

**Reply by ImEdmond** [reply]
@nbrown02 Thank you for the fast reply. Yes I´m using Jira Cloud.

**Reply by nbrown02** [reply]
For Jira I don't know how to get the iterations data for issues I'm afraid :(

**Comment by DrFish22** [comment]
Managed to test it out on a different project.  Looks like filter doesn't like the switch to only showing iterations! If there is a way I can ix that then just let me know as it would be great if we could pick how many iterations to show across the 4 charts. Other than that this is exactly what I'm looking for, the team thinks and works in Sprints not weeks, so this makes this much more likely to be adoptable by the teams.

![image](https://user-images.githubusercontent.com/15156696/147094160-d36efc26-1b2c-4f99-9f46-cf83d0b8af47.png)

**Reply by nbrown02** [reply]
Ah I think I know why that may be, no relationship between dates table and iterations table I think - let me do some quick tweaks :)

**Reply by nbrown02** [reply]
So the only way this is possible (I think) is by bringing EndDate (or StartDate) into your date range filter (you can rename it to be 'Range' if you like)...albeit my dataset is limited!

<img width="857" alt="Screenshot 2021-12-22 at 14 01 53" src="https://user-images.githubusercontent.com/29369962/147104554-d4f8d26b-3217-454f-80b0-49fa6a6428ea.png">

Can you give that a try?

**Reply by DrFish22** [reply]
Managed to add an iteration filter, then just select the iterations I want to see. Works perfectly I think.

Got some odd results for the sustainability (net flow) chart which I will dig into a bit more and compare with some other charting/reporting we have been looking at. 

Looks really good.

Cheers.

**Reply by nbrown02** [reply]
Thank you - if it's useful please don't hesitate to leave us a review on the [Azure DevOps marketplace](https://marketplace.visualstudio.com/items?itemName=agile-extensions.flowviz#review-details) :)

**Comment by DrFish22** [comment]
Finally getting to look at this again. Got a few issues. Looks like I have problems on a couple of the charts when there are multiple teams set upin ADO.

![image](https://user-images.githubusercontent.com/15156696/153575592-8fbf44be-b2e5-4821-a599-a6f211967387.png)
Also, can I alter the "bugs" it is detecting, I think we might have a new type defined for some projects that the graph looks like its not finding. Whether that is the cause of the NaN on both the quality and flow charts I am not sure.

**Reply by nbrown02** [reply]
Thanks for getting back to me, so things I can see:
- Net flow chart not working (reason unclear)
- Quality chart not working (potential reason being different work item type for 'bug')

Is that a fair summary? 

For bug rate, you should be able to change the work item type in the 'CompletedBugs' column, highlighted here

<img width="813" alt="Screenshot 2022-02-11 at 10 54 36" src="https://user-images.githubusercontent.com/29369962/153579538-c4bd7b3a-8355-4449-9fc8-9a1dd2215a83.png">

**Comment by DrFish22** [comment]
Hi
Yes and the duplicate sprints showing up despite filtering by team.
Cheers
Chris

Sent from Yahoo Mail on Android 
 
  On Fri, 11 Feb 2022 at 10:55 am, Nick ***@***.***> wrote:   


Thanks for getting back to me, so things I can see:
   
   - Net flow chart not working (reason unclear)
   - Quality chart not working (potential reason being different work item type for 'bug')

Is that a fair summary?

For bug rate, you should be able to change the work item type in the 'CompletedBugs' column, highlighted here



—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by nbrown02** [comment]
Could you share what your iterations table looks like?

**Comment by DrFish22** [comment]
Hi
This is real data I havent sent you obviosuly in case anyone asks ;)


Does that show what you need? 
Chris

    On Friday, 11 February 2022, 11:51:29 GMT, Nick Brown ***@***.***> wrote:  
 
 


Could you share what your iterations table looks like?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by nbrown02** [comment]
I'm wondering if any of your sprints for that team have changed dates and that could be why there are duplicate entries...

**Comment by DrFish22** [comment]
Hi
Staring at the data shows it looks like duplicate names. If I switch to the full iteration path for this data it looks like it might function better. Will keep playing and let you know. Also going to try and get the bug filters working with our odd extra types.
Thanks
Chris
    On Friday, 11 February 2022, 12:05:03 GMT, Nick Brown ***@***.***> wrote:  
 
 


I'm wondering if any of your sprints for that team have changed dates and that could be why there are duplicate entries...

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by nbrown02** [comment]
Cool thank you - I'm going to have a play around too and see if there is anything else I can spot/fix :)

**Comment by nbrown02** [comment]
You can add `&$expand=Teams($select=TeamName)` to the end of the iterations query, mimic the steps from the Areas table to expand the column and get the team names there. Not sure if that helps:

<img width="1244" alt="Screenshot 2022-02-11 at 12 55 27" src="https://user-images.githubusercontent.com/29369962/153595684-c620aa2a-3bbf-4b04-8743-c7167622d002.png">

For Net Flow I really am unsure why that doesn't work...

**Comment by DrFish22** [comment]
Hi
Thanks, still playing with the data from the project with multiple teams. Thought I would try the two other main projects we need data for and in both cases i'm getting a cyclic dependency error trying to import data.
Is there any way I can identify what data is at fault so I can fix it?
Cheers
Chris
    On Friday, 11 February 2022, 13:00:21 GMT, Nick Brown ***@***.***> wrote:  
 
 


You can add &$expand=Teams($select=TeamName) to the end of the iterations query, mimic the steps from the Areas table to expand the column and get the team names there. Not sure if that helps:



For Net Flow I really am unsure why that doesn't work...

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by nbrown02** [comment]
Hmm - whenever I’ve seen that, a refresh solves it. When I googled it I didn’t find the guidance particularly helpful…
